multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/util/ClientKeyConfigurationHandler.java

```diff
@@ -0,0 +1,32 @@
+package org.cloudfoundry.multiapps.controller.persistence.util;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.cloudfoundry.multiapps.controller.persistence.Messages;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClientKeyConfigurationHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientKeyConfigurationHandler.class);
+
+    public void createEncodedKeyFile(String pemKey, String fileName) {
+        try (PEMParser pemParser = new PEMParser(new StringReader(pemKey))) {
+            PEMToEncodedKeyConverter derKeyHandler = new PEMToEncodedKeyConverter();
+            JcaPEMKeyConverter converter = new JcaPEMKeyConverter();
+            byte[] encodedKey = derKeyHandler.getPrivateEncodedKey(pemParser, converter);
+            writeEncodedKeyToFile(encodedKey, fileName);
+        } catch (IOException e) {
+            LOGGER.error(MessageFormat.format(Messages.GENERATING_KEY_FILE_FAILED, e.getMessage()));
```

Comment:
Shouldn't we rethrow the error?

Reply:
I agree the exception should be changed to be rethrown.